tests/unit/test_container_to_args_secrets.py

```diff
@@ -89,3 +92,317 @@ async def test_pass_secret_as_env_variable_no_external(self):
+        c = create_compose_mock()
+        c.declared_secrets = {
+            "my_secret_name": {
+                "external": "true",
+            }
+        }
+        cnt = get_minimal_container()
+        cnt["_service"] = "test-service"
+        cnt["secrets"] = [
+            {"source": "my_secret_name", "target": "my_secret_name", "type": "does_not_equal_env"}
+        ]
+
+        with self.assertLogs() as cm:
+            args = await container_to_args(c, cnt)
+        self.assertEqual(len(cm.output), 1)
+        self.assertEqual('That is un-supported and a no-op and is ignored.' in cm.output[0], True)
+        self.assertEqual('my_secret_name' in cm.output[0], True)
+
+        self.assertEqual(
+            args,
+            [
+                "--name=project_name_service_name1",
+                "-d",
+                "--network=bridge",
+                "--network-alias=service_name",
+                "--secret",
+                "my_secret_name,type=does_not_equal_env",
+                "busybox",
+            ],
+        )
+
+    @parameterized.expand([
+        (
+            "no_secret_target",
+            {
+                "file_secret": {
+                    "file": "./my_secret",
+                }
+            },
+            "file_secret",
+            "/test_dirname/my_secret:/run/secrets/file_secret:ro,rprivate,rbind",
+        ),
+        (
+            "custom_target_name",
+            {
+                "file_secret": {
+                    "file": "./my_secret",
+                }
+            },
+            {
+                "source": "file_secret",
+                "target": "custom_name",
+            },
+            "/test_dirname/my_secret:/run/secrets/custom_name:ro,rprivate,rbind",
+        ),
+        (
+            "no_custom_target_name",
+            {
+                "file_secret": {
+                    "file": "./my_secret",
+                }
+            },
+            {
+                "source": "file_secret",
+            },
+            "/test_dirname/my_secret:/run/secrets/file_secret:ro,rprivate,rbind",
+        ),
+        (
+            "custom_location",
+            {
+                "file_secret": {
+                    "file": "./my_secret",
+                }
+            },
+            {
+                "source": "file_secret",
+                "target": "/etc/custom_location",
+            },
+            "/test_dirname/my_secret:/etc/custom_location:ro,rprivate,rbind",
+        ),
+    ])
+    async def test_file_secret(
+        self, test_name, declared_secrets, add_to_minimal_container, volume_ref
+    ):
+        c = create_compose_mock()
+        c.declared_secrets = declared_secrets
+        cnt = get_minimal_container()
+        cnt["_service"] = "test-service"
+        cnt["secrets"] = [add_to_minimal_container]
+        args = await container_to_args(c, cnt)
+        base_dir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
```

Comment:
Extract to separate function because it's likely to have repeat uses (at least that's the case in other tests).